mesonbuild/dependencies/misc.py

```diff
@@ -175,7 +175,7 @@ def __init__(self, name: str, environment: 'Environment', kwargs: T.Dict[str, T.
             return
 
         self.name = 'python3'
-        self.static = kwargs.get('static', False)
+        self.static = kwargs.get('static', environment.coredata.get_option(mesonlib.OptionKey('prefer_static')))
```

Comment:
This seems odd! Why does this line exist, it seems to be literally overriding the existing self.static attribute...

Reply:
Do you mean that this is attribute is already inherited from `base.py`? Yeah, I can just delete the line in that case.

Reply:
That is exactly what I mean. :)